provider/lxd/config.go

```diff
@@ -215,11 +215,7 @@ func (c *environConfig) clientConfig() (lxdclient.Config, error) {
 	cfg := lxdclient.Config{
 		Remote: remote,
 	}
-	cfg, err := cfg.WithDefaults()
-	if err != nil {
-		return cfg, errors.Trace(err)
-	}
-	return cfg, nil
+	return cfg.WithDefaults()
```

Comment:
every such short-circuit makes me have to stop and double-check that there's no opportunity for hidden-nils to creep in; even though this one *is* safe, it's annoying to have to stop and check what *really* happens when you use that construct. revert?